### PR TITLE
Make project Mbed OS 6 ready

### PR DESCRIPTION
**Pull request by hugueskamba**
* Reference mbed-os-6.0.0-beta-1 in mbed-os.lib
* Amend README to reflect behaviors of Mbed OS 6 and its build tool
* Add small C library configuration to application configuration file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,15 +76,15 @@ To build with the smaller C libraries, modify the application configuration file
 }
 ```
 
-If the build system throws an error when compiling with the small C library, you can find more information [here]( https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/program-setup/bare_metal/c_small_libs.md).
+The build system build system reverts to the standard C library if support for the small C library is not enabled for your target. You can find more information [here]( https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/program-setup/bare_metal/c_small_libs.md).
```
Comment (evedon) [inline README.md:79]: Could you link to https://os.mbed.com/docs/mbed-os/v6.0-preview/bare-metal/using-small-c-libraries.html
@iriark01 what branch should we use?

Comment (iriark01) [reply README.md:79]: For now use v6.0-preview, as /latest/ doesn't redirect to it (because it's not official, so /latest/ goes to 5.15)

Comment (evedon) [reply README.md:79]: but we don't really want to update the link...

Comment (evedon) [reply README.md:79]: Perhaps better to leave the link to github then

**Review by evedon — APPROVED** [review]

**Pull request merged**